standard/attributes.md

```diff
@@ -1060,6 +1060,44 @@ Specifies that a nullable argument won’t be `null` when the method returns the
 >
 > *end example*
 
+### §enumerator-cancellation The EnumeratorCancellation attribute
+
+Specifies the parameter representing the `CancellationToken` for an asynchronous iterator (§15.15). The value of this parameter is a combination of two cancellation tokens: the token which was actually passed to the method, and the token passed to the `IAsyncEnumerable<T>.GetAsyncEnumerator(CancellationToken)` implementation on the generated iterator. The combined token will be canceled if either of the two source tokens are canceled.
+
+It is an error if the `EnumeratorCancellation` attribute is applied to more than one parameter. The compiler may produce a warning if the `EnumeratorCancellation` attribute is applied to a parameter of a different type than `CancellationToken` or on a method that isn't an asynchronous iterator (§15.15) which returns an asynchronous enumerable interface ([§15.15.3](classes.md#15153-enumerable-interfaces)) rather than an asynchronous enumerator interface ([§15.15.2](classes.md#15152-enumerator-interfaces)). If no attributes have this parameter, the iterator won't have access to the `CancellationToken` argument for `GetAsyncEnumerator`.
```

Comment:
The second half of this sentence is a bit hard to read - the "or on a method that isn't an asynchronous iterator which returns an asynchronous enumerable interface rather than an asynchronous enumerator interface" part.

I think it would be easier to read if we just removed "rather than an asynchronous enumerator interface".

Alternatively, we could change it to a set of bullet points, for situations of:

- The attribute is applied to the wrong type of parameter
- The attribute is applied on a method which isn't an asynchronous iterator
- The attribute is applied on a method which returns an asynchronous enumerator instead of an asynchronous enumerable interface

(But I'd prefer just deleting the "rather than an asynchonrous enumerator interface" part.)